### ISSUE TITLE
sanitizer.Sanitize may cause panic

### ISSUE THREAD

**Issue by wzy9607** [bugsnag/bugsnag-go#222]
### Describe the bug

sanitizer.Sanitize may cause panic by calling value method using nil pointer.

### Steps to reproduce

1. Run the Example code snippet
2. See a panic

### Environment
* Bugsnag Go version: 2.3.0
* Go version: 1.22.1
* Integration framework version:
    * Martini:
    * Negroni:
    * net/http:
    * Revel:
    * Other:

### Example code snippet

```go
package main

import (
	"errors"

	bugsnago "github.com/bugsnag/bugsnag-go/v2"
)

type Name struct {
	name string
}

func (n Name) String() string {
	return n.name
}

type data struct {
	Name *Name
}

func main() {
	err := errors.New("error")
	meta := bugsnago.MetaData{}
	meta.AddStruct("data", map[string]any{
		"data": data{Name: nil},
	})
	_ = bugsnago.Notify(err, meta)
}
```

<details><summary>Error messages:</summary>

```
panic: value method main.Name.String called using nil *Name pointer

goroutine 1 [running]:
main.(*Name).String(0xc00006c0e0?)
        <autogenerated>:1 +0x25
github.com/bugsnag/bugsnag-go/v2.sanitizer.Sanitize({{0x0, 0x0, 0x0}, {0xc00002e240, 0x3, 0x4}}, {0xec50fc0, 0x0})
        /Users/user/go/pkg/mod/github.com/bugsnag/bugsnag-go/v2@v2.3.0/metadata.go:98 +0x7b
...
...
...
```
</details>

**Comment by wzy9607** [comment]
I see there is already a pr fixing this, so I'll close this issue.

**Comment by DariaKunoichi** [comment]
@wzy9607 Hey, the fix is merged and released in v2.3.1.
